src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Debugging/DefaultRazorBreakpointResolver.cs

```diff
@@ -70,6 +72,11 @@ internal class DefaultRazorBreakpointResolver : RazorBreakpointResolver
             _documentMappingProvider = documentMappingProvider;
             _csharpBreakpointResolver = csharpBreakpointResolver;
             _workspace = workspace;
+
+            // 4 is a magic number that was determined based on the functionality of VisualStudio. Currently when you set or edit a breakpoint
+            // we get called with two different locations for the same breakpoint. Because of this 2 time call our size must be at least 2,
+            // we grow it to 4 just to be safe for lesser known scenarios.
+            _cache = new MemoryCache<CacheKey, Range>(sizeLimit: 4);
```

Comment:
Oh no, my crimes are propagating! But for real it would be nice if we could kill this class and use the one from `Microsoft.Extensions.Caching.Memory`, I'm sure it's more performant than my implementation by a good bit.

Reply:
Only problem if we do that is that we'll start loading another assembly which we don't want to 😢